### PR TITLE
Migrate integration tests to pytest

### PR DESCRIPTION
**Pull request by dirkkul**
- Moves all integration tests to pytest
- Drops python 3.6

### CODE REVIEW

File: integration/test_backup.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        client.data_object.create(art["properties"], "Article", art["id"])
+        client.data_object.reference.add(
+            from_uuid=art["id"],
+            from_class_name="Article",
+            from_property_name="hasParagraphs",
+            to_uuid=paragraphs[i]["id"],
+            to_class_name="Paragraph"
+        )
+    yield client
+    for _, cls in enumerate(schema["classes"]):
```
[the remark this answers is not in the document]
Comment (StefanBogdan) [reply integration/test_backup.py:137]: Also here we could just call `client.schema.delete_all()` which does exactly this (but first gets the schema from Weaviate).

File: integration/test_crud.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+
+@pytest.fixture(scope="module")
+def client(people_schema):
+    client = weaviate.Client("http://localhost:8080")
+    client.schema.create(people_schema)
+
+    yield client
+    for cls in people_schema["classes"]:
+        client.schema.delete_class(cls["class"])
```
Comment (StefanBogdan) [inline integration/test_crud.py:81]: Same here with `client.schema.delete_all()`.

File: integration/test_graphql.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    client.data_object.create({"name": "B", "size": 20}, "Ship")
+    client.data_object.create({"name": "C", "size": 43}, "Ship")
+    client.data_object.create({"name": "D", "size": 1}, "Ship")
+    client.data_object.create({"name": "E", "size": 34}, "Ship")
+    client.data_object.create({"name": "F", "size": 303}, "Ship")
+    time.sleep(2.0)
+
+    yield client
+    for _, cls in enumerate(schema["classes"]):
+        client.schema.delete_class(cls["class"])
```
Comment (StefanBogdan) [inline integration/test_graphql.py:46]: Same here with `client.schema.delete_all()`

File: integration/test_nodes.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+
+@pytest.fixture(scope="module")
+def client():
+    client = weaviate.Client("http://localhost:8080")
+
+    yield client
+    for _, cls in enumerate(schema["classes"]):
+        client.schema.delete_class(cls["class"])
```
Comment (StefanBogdan) [inline integration/test_nodes.py:41]: Same here with `client.schema.delete_all()`

File: .github/workflows/tests.yaml [hunk 5]
```diff
@@ -14,12 +20,19 @@ jobs:
         cache: 'pip' # caching pip dependencies
     - run: pip install -r requirements.txt
     - name: Run unittests
-      run: python -m unittest -v
+      run: pytest --cov --cov-report=term-missing --cov=weaviate --cov-report xml:coverage.xml test
+    - name: Upload to codecov
```
Comment (StefanBogdan) [inline .github/workflows/tests.yaml:24]: Does this publish the results somewhere? Do we need an account for this?

File: .codecov.yml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+codecov:
```
Comment (StefanBogdan) [inline .codecov.yml:1]: This tool is new for me so I do not know exactly how it should be configured.

File: integration/test_backup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dirkkul) [reply integration/test_backup.py:137]: also, copy & paste, removed everywhere :)

File: .github/workflows/tests.yaml (hunk 5 above)
Comment (dirkkul) [reply .github/workflows/tests.yaml:24]: We don't need an account, the results are added [as a reply to the PR](https://github.com/semi-technologies/weaviate-python-client/pull/137#issuecomment-1281898094).
(not the biggest fan of this, but its how it's done in weaviate-core and I just copied everything over)

Comment (StefanBogdan) [reply .github/workflows/tests.yaml:24]: Nice. Thanks for the info.

**Review by StefanBogdan — APPROVED** [review]

**Pull request merged**